wgpu/src/lib.rs

```diff
@@ -3887,7 +3901,17 @@ impl CommandEncoder {
     /// Begins recording of a compute pass.
     ///
     /// This function returns a [`ComputePass`] object which records a single compute pass.
-    pub fn begin_compute_pass(&mut self, desc: &ComputePassDescriptor<'_>) -> ComputePass {
+    ///
+    /// As long as the returned  [`ComputePass`] has not ended,
+    /// any mutable operation on this command encoder causes an error and invalidates it.
+    /// Note that the lifetime constraint protects against this, but it is possible to opt out of it
+    /// by calling [`ComputePass::make_static`].
+    /// This can be useful for runtime handling of the encoder->pass
+    /// dependency e.g. when pass and encoder are stored in the same data structure.
+    pub fn begin_compute_pass<'a>(
+        &'a mut self,
+        desc: &ComputePassDescriptor<'_>,
+    ) -> ComputePass<'a> {
```

Comment:
For consistency:

```suggestion
    pub fn begin_compute_pass<'pass>(
        &'pass mut self,
        desc: &ComputePassDescriptor<'_>,
    ) -> ComputePass<'pass> {
```

Reply:
makes sense, but I'll go with `'encoder` since really that's the lifetime we're tracking here